### PR TITLE
Initialising BinaryLabels & MulticlassLabels with probabilities scores

### PR DESCRIPTION
**Pull request by olinguyen**
- BinaryLabels constructor with confidence scores is more intuitive (comparing >= threshold)
- Added MulticlassLabels constructor with probability matrix

### CODE REVIEW

File: src/shogun/labels/MulticlassLabels.cpp [hunk 1]
```diff
@@ -20,6 +20,28 @@ CMulticlassLabels::CMulticlassLabels(const SGVector<float64_t> src) : CDenseLabe
 	set_labels(src);
 }
 
+CMulticlassLabels::CMulticlassLabels(const SGMatrix<float64_t> src)
```
Comment (karlnapf) [inline src/shogun/labels/MulticlassLabels.cpp]: `src` is not a good name, a better one would be `scores`

File: src/shogun/labels/MulticlassLabels.cpp [hunk 2]
```diff
@@ -20,6 +20,28 @@ CMulticlassLabels::CMulticlassLabels(const SGVector<float64_t> src) : CDenseLabe
...
+    : CDenseLabels()
+{
+	init();
+	int32_t n_classes = src.num_cols;
+
+	auto labels = SGVector<float64_t>(src.num_rows);
```
Comment (karlnapf) [inline src/shogun/labels/MulticlassLabels.cpp]: somehow cleaner is: `SGVector<float64_t> labels(src.num_rows)`

File: src/shogun/labels/MulticlassLabels.cpp (hunk 1 above)
Comment (karlnapf) [inline src/shogun/labels/MulticlassLabels.cpp]: or confidences

File: src/shogun/labels/MulticlassLabels.h [hunk 3]
```diff
@@ -53,6 +53,12 @@ class CMulticlassLabels : public CDenseLabels
 
 		/** constructor
 		 *
+		 * @param src labels to set from a probability matrix
```
Comment (karlnapf) [inline src/shogun/labels/MulticlassLabels.h]: this is not necessarily a probability matrix right? Can be arbritary scores

Comment (karlnapf) [inline src/shogun/labels/MulticlassLabels.h]: You could say from a matrix that contains scores for each class for each point. Those could be probabilities but do not have to be normalized

File: tests/unit/labels/BinaryLabels_unittest.cc [hunk 4]
```diff
@@ -66,4 +86,27 @@ TEST(BinaryLabels, serialization)
 		EXPECT_NEAR(labels->get_label(i), new_labels->get_label(i), 1E-15);
 	}
 	unlink(filename);
+
+	SG_UNREF(labels);
+	SG_UNREF(new_labels);
+}
+
+TEST_F(BinaryLabels, set_values_labels_from_constructor)
```
Comment (karlnapf) [inline tests/unit/labels/BinaryLabels_unittest.cc]: good!

File: tests/unit/labels/BinaryLabels_unittest.cc [hunk 5]
```diff
@@ -66,4 +86,27 @@ TEST(BinaryLabels, serialization)
...
+{
+	CBinaryLabels* labels = new CBinaryLabels(probabilities);
+
+	SGVector<float64_t> labels_vector = labels->get_labels();
+	SGVector<float64_t> values_vector = labels->get_values();
+
```
Comment (karlnapf) [inline tests/unit/labels/BinaryLabels_unittest.cc]: you could put `ASSERT` here for the pointer not being NULL, and for the size of the vectors to be both equal to N

File: tests/unit/labels/MulticlassLabels_unittest.cc [hunk 6]
```diff
@@ -43,3 +75,24 @@ TEST(MulticlassLabelsTest,confidences)
 	}
 	SG_UNREF(labels);
 }
+
+TEST_F(MulticlassLabelsTest, prob_matrix_label_initialization)
```
Comment (karlnapf) [inline tests/unit/labels/MulticlassLabels_unittest.cc]: as said, this dont have to be probabilities

File: src/shogun/labels/BinaryLabels.cpp [hunk 7]
```diff
@@ -54,7 +54,7 @@ CBinaryLabels::CBinaryLabels(SGVector<float64_t> src, float64_t threshold) : CDe
 	SGVector<float64_t> labels(src.vlen);
 	for (int32_t i = 0; i < labels.vlen; i++)
 	{
-		labels[i] = src[i] + threshold >= 0 ? +1.0 : -1.0;
+		labels[i] = src[i] >= threshold ? +1.0 : -1.0;
```
Comment (karlnapf) [inline src/shogun/labels/BinaryLabels.cpp]: can you check whether and where this constructor is called. You are changing behaviour of existing code here, so need to account for that

Comment (olinguyen) [inline src/shogun/labels/BinaryLabels.cpp]: It is called in apply_binary, but I set the default threshold value to 0, as it initially was, so it shouldn't be affecting the existing code.

Comment (karlnapf) [inline src/shogun/labels/BinaryLabels.cpp]: ok!

**Comment by karlnapf** [comment]
btw you committed changes to the `data` submodule and this is why the tests fail.
Needs to be removed before a merge

**Comment by karlnapf** [comment]
Ok lets wait for travis and then merge

**Pull request merged**